internal/metamorph/zmq.go

```diff
@@ -181,6 +155,53 @@ func (z *ZMQ) Start(zmqi ZMQI) error {
 	return nil
 }
 
+func (z *ZMQ) handleInvalidTx(msg []string) (hash *chainhash.Hash, status metamorph_api.Status, txErr error, competingTxs []string, err error) {
+	txInfo, err := z.parseTxInfo(msg)
+	if err != nil {
+		return
+	}
+
+	hash, err = chainhash.NewHashFromStr(txInfo.TxID)
+	if err != nil {
+		return
+	}
+
+	if txInfo.IsMissingInputs {
+		// Missing Inputs does not immediately mean it's an error. It may mean that transaction is temporarily waiting
+		// for its parents (e.g. in case of bulk submit). So we don't throw any error here, just update the status.
+		// If it's actually an error with transaction, it will be rejected when the parents arrive to node's memopool.
+		// If the parents never arrive, it will be discarded from mempool after some time - rejected.
+		status = metamorph_api.Status_SEEN_IN_ORPHAN_MEMPOOL
+		return
+	}
+
+	if txInfo.IsMempoolConflictDetected || txInfo.IsDoubleSpendDetected || len(txInfo.CollidedWith) > 0 {
+		// In case of Double Spend Attempted, we don't want to add an error to the response, as it will immediately
+		// return this response to the user. We want to inform him about the status update by callback and wait,
+		// because the node can accept one of the transactions from double spend and reject the other(s). Then,
+		// we will update the other transactions with the REJECTED status and an error.
+		status = metamorph_api.Status_DOUBLE_SPEND_ATTEMPTED
+		for _, ctx := range txInfo.CollidedWith {
```

Comment:
Please rename `ctx` to `cTx`/`tx` or something since ctx usually means "context"

Reply:
Right, done